src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs

```diff
@@ -0,0 +1,208 @@
+            {
+                Array returnValueArray = returnValue as Array;
+                if (returnValueArray.Length == 1)
+                {
+                    object unitTestResultObject = returnValueArray.GetValue(0);
+                    if (unitTestResultObject != null)
+                    {
+                        UnitTestResultStruct unitTestResult = unitTestResultObject.As<UnitTestResultStruct>();
+
+                        switch (unitTestResult.Outcome)
+                        {
+                            case UnitTestResultOutcome.Error:
+                            case UnitTestResultOutcome.Failed:
+                            case UnitTestResultOutcome.NotFound:
+                            case UnitTestResultOutcome.Timeout:
+                                scope.Span.SetTag(TestTags.Status, TestTags.StatusFail);
+                                scope.Span.Error = true;
+                                scope.Span.SetTag(Tags.ErrorMsg, unitTestResult.ErrorMessage);
+                                scope.Span.SetTag(Tags.ErrorStack, unitTestResult.ErrorStackTrace);
+                                break;
+                            case UnitTestResultOutcome.Inconclusive:
+                            case UnitTestResultOutcome.NotRunnable:
+                            case UnitTestResultOutcome.Ignored:
+                                scope.Span.SetTag(TestTags.Status, TestTags.StatusSkip);
+                                scope.Span.SetTag(TestTags.SkipReason, unitTestResult.ErrorMessage);
+                                break;
+                            case UnitTestResultOutcome.Passed:
+                                scope.Span.SetTag(TestTags.Status, TestTags.StatusPass);
+                                break;
+                        }
+                    }
+                }
+
+                scope.Dispose();
+            }
+
+            return new CallTargetReturn<TReturn>(returnValue);
+        }
+
+        private static Dictionary<string, List<string>> GetTraits(MethodInfo methodInfo)
+        {
+            Dictionary<string, List<string>> testProperties = null;
+            var testAttributes = methodInfo.GetCustomAttributes(true);
+            foreach (var tattr in testAttributes)
+            {
+                if (tattr?.GetType().Name != "TestCategoryAttribute")
+                {
+                    continue;
+                }
+
+                testProperties ??= new Dictionary<string, List<string>>();
+                if (!testProperties.TryGetValue("Category", out var categoryList))
+                {
+                    categoryList = new List<string>();
+                    testProperties["Category"] = categoryList;
+                }
+
+                categoryList.AddRange(tattr.As<TestCategoryAttributeStruct>().TestCategories);
+            }
+
+            var classCategories = methodInfo.DeclaringType?.GetCustomAttributes(true);
+            if (!(classCategories is null))
```

Comment:
Note that you can use `if (classCategories is not null)` now

Reply:
changed in f34a8cd